spec/Classes.tex

```diff
@@ -784,12 +787,54 @@ \subsection{The Method Receiver and the {\em this} Argument}
 is \chpl{c1}.
 \end{chapelexample}
 
-The optional \sntx{this-intent} declares the intent by which the receiver
-argument will be passed.  When the value of \sntx{this-intent} is \chpl{param},
-it implies that the function can only be applied to param objects of the given
-type binding.  If the value is \chpl{ref}, the argument will be passed by
-reference---allowing modification of \chpl{this}.  Otherwise, the argument will
-be passed with the blank intent as specified in \rsec{The_Blank_Intent}.
+The optional \sntx{this-intent} declares how the receiver argument is
+constrained or will be passed for a given method call.
+
```

Comment:
To make sure I'm understanding correctly, I think you're saying:

* In Smalltalk, Self, and CLOS, if a method is defined on a class (rather than an instance of that class), it can only be called on the class, not on instances of that class.
* The currently-implemented behavior of Chapel is therefore closer to Smalltalk, Self, and CLOS
* However, in most other contexts, Chapel is more like C++/Java than Smalltalk/Self/CLOS. Therefore we may want to choose not to differ from them in this regard (suggesting the implementation would need to change).

Did I get that right?